hwy/tests/mul_test.cc

```diff
@@ -743,6 +839,97 @@ HWY_NOINLINE void TestAllRearrangeToOddPlusEven() {
   ForShrinkableVectors<TestRearrangeToOddPlusEven>()(bfloat16_t());
 #endif
   ForShrinkableVectors<TestRearrangeToOddPlusEven>()(int16_t());
+  ForShrinkableVectors<TestRearrangeToOddPlusEven>()(uint16_t());
+}
+
+template <bool MixedSignedness>
+struct TestSumOfMulQuadAccumulate {
+  template <class DW2, class TN1, class TN2>
+  static HWY_INLINE void TestConsecutiveSeqMulQuadAccum(DW2 dw2, TN1 a0,
+                                                        TN2 b0) {
+    using TW2 = TFromD<DW2>;
+    const Repartition<TN1, DW2> dn1;
+    const Repartition<TN2, DW2> dn2;
+
+    const auto vn_iota0_mod4 = And(Iota(dn1, TN1{0}), Set(dn1, TN1{3}));
+
+    const auto va = Add(vn_iota0_mod4, Set(dn1, a0));
+    const auto vb = Add(BitCast(dn2, vn_iota0_mod4), Set(dn2, b0));
+    const auto expected =
+        Set(dw2,
+            static_cast<TW2>((TW2{4} * static_cast<TW2>(a0) * b0) +
+                             (TW2{6} * (static_cast<TW2>(a0) + b0)) + TW2{17}));
+
+    HWY_ASSERT_VEC_EQ(dw2, expected,
+                      SumOfMulQuadAccumulate(dw2, va, vb, Set(dw2, TW2{3})));
+  }
+
+  template <typename TN2, class DN2>
+  HWY_INLINE void operator()(TN2 /*unused*/, DN2 dn2) {
+    static_assert(!MixedSignedness || IsSigned<TN2>(),
+                  "TN2 must be signed if MixedSignedness is true");
+    using TN1 = If<MixedSignedness, MakeUnsigned<TN2>, TN2>;
+    using TW2 = MakeWide<MakeWide<TN2>>;
+
+    const Rebind<TN1, DN2> dn1;
+    const Repartition<TW2, DN2> dw2;
+
+    const auto vn1_k1 = Set(dn1, TN1{1});
+    const auto vn2_k1 = BitCast(dn2, vn1_k1);
+    const auto vn1_k4 = Set(dn1, TN1{4});
+    const auto vn2_k4 = BitCast(dn2, vn1_k4);
+
+    const auto vw2_k0 = Zero(dw2);
+    const auto vw2_k1 = Set(dw2, TW2{1});
+    const auto vw2_k4 = Set(dw2, TW2{4});
+    const auto vw2_k5 = Set(dw2, TW2{5});
+    const auto vw2_k21 = Set(dw2, TW2{21});
+
+    HWY_ASSERT_VEC_EQ(dw2, vw2_k4,
+                      SumOfMulQuadAccumulate(dw2, vn1_k1, vn2_k1, vw2_k0));
+    HWY_ASSERT_VEC_EQ(dw2, vw2_k4,
```

Comment:
Looks like these lines are identical - did we intend something else, or should we remove?

Reply:
Removed the duplicate line